### PR TITLE
Add order as Apollo Federation entity

### PR DESCRIPTION
**Pull request by binary-koan**
I want to merge this change because I want to build custom APIs that return orders with Apollo Federation. Currently it is possible to extend the Order type in a federated API, but it is not possible to return an Order from a custom type because it's not a federated entity.

I tried to be somewhat consistent with how Product entities work, and hopefully figured out strict enough permission handling. Wasn't sure exactly what to test, but I tried to cover a reasonable number of cases.

I wasn't sure whether to make `channel` a key; since it works differently in orders compared to products etc I figured it wasn't worth the effort, but I could probably figure out how to make a key of `id channel.id` if that's a better solution.

Happy to make any changes to get this mergeable - we'd really like to improve our federated API with this change!

Partially addresses #11986

# Impact

* [ ] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [x] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/order/resolvers.py [hunk 1]
```diff
@@ -9,15 +9,25 @@
...
         qs = qs.filter(channel__slug=str(channel_slug))
+    if ids:
+        db_ids = [
+            from_global_id_or_error(node_id, "Order")[1] for node_id in ids
+        ]
+        qs = qs.filter(pk__in=db_ids)
+    if requesting_user and not requestor_has_access_to_all:
+        qs = qs.filter(user_id=requesting_user.id)
+    if not requestor_has_access_to_all:
+        qs = qs.exclude(status=OrderStatus.DRAFT)
```
[the remark this answers is not in the document]
Comment (binary-koan) [reply saleor/graphql/order/resolvers.py]: Oh yeah, true 🤦 I wonder if draft orders should be returned as entities, but we don't have a use case for it anyway

Also fixed the tests, that was just a silly mistake when I extended the resolver

File: saleor/graphql/order/types.py [hunk 2]
```diff
@@ -1822,6 +1825,34 @@ def resolve_checkout_id(root: models.Order, _info):
...
+            )
+        else:
+            qs = models.Order.objects.none()
+
+        orders = {}
+        for order in qs:
+            global_id = graphene.Node.to_global_id("Order", order.id)
+            orders[global_id] = order
+
+        return [orders.get(root_id) for root_id in roots_ids]
```
Comment (maarcingebala) [inline saleor/graphql/order/types.py:1854]: We have `resolve_federation_references` helper function which solves this part, I think we could reuse it here.

Comment (maarcingebala) [reply saleor/graphql/order/types.py:1854]: Other than that it looks good IMO :)

**Comment by maarcingebala** [comment]
@binary-koan There are some conflicts after changes [these changes](https://github.com/saleor/saleor/pull/12339) were merged, I believe, could you take a look? Also, CI is failing - we use Precommit which helps with [running the same checks locally](https://docs.saleor.io/docs/3.x/developer/community/contributing#coding-style), I think you need to just fix the import order and generate new GraphQL schema file (`npm run build-schema`).

**Review by korycins — APPROVED** [review]

**Comment by maarcingebala** [comment]
@binary-koan Looks good; one last missing thing - you can add yourself to the changelog file. In the [3.13 > GraphQL section](https://github.com/saleor/saleor/blob/main/CHANGELOG.md#graphql-api) add an entry with the description, PR number and your GH nickname. It will be included in the release notes for Saleor 3.13 :)

**Review by maarcingebala — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Comment by binary-koan** [comment]
@maarcingebala Sweet, done! Thanks for the reviews!

**Pull request merged**